daemon/mgr/image_load.go

```diff
@@ -0,0 +1,50 @@
+package mgr
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/alibaba/pouch/pkg/reference"
+
+	ociimage "github.com/containerd/containerd/images/oci"
+	pkgerrors "github.com/pkg/errors"
+)
+
+// LoadImage loads images by the oci.v1 format tarstream.
+func (mgr *ImageManager) LoadImage(ctx context.Context, imageName string, tarstream io.ReadCloser) error {
+	defer tarstream.Close()
+
+	namedRef, err := reference.Parse(imageName)
+	if err != nil {
+		return pkgerrors.Wrapf(err, "failed to parse image name %s", imageName)
+	}
+
+	// NOTE: in the image ocispec.v1, the org.opencontainers.image.ref.name
+	// annotation represents a "tag" for image. For example, an image may
+	// have a tag for different versions or builds of the software.
+	// And containerd.importer will append ":" and annotation so that we
+	// don't allow imageName to contains any digest or tag information.
+	// We don't want to see the any reference like foo/bar:latest:v1.2.
+	if !reference.IsNamedOnly(namedRef) {
+		return fmt.Errorf("the image name should not contains any digest or tag information")
+	}
+
+	importer := &ociimage.V1Importer{
+		ImageName: imageName,
+	}
+
+	imgs, err := mgr.client.ImportImage(ctx, importer, tarstream)
+	if err != nil {
+		return pkgerrors.Wrap(err, "failed to import image into containerd by tarstream")
+	}
+
+	// FIXME(fuwei): if the store fails to update reference cache, the daemon
+	// may fails to restart.
+	for _, img := range imgs {
+		if err := mgr.storeImageReference(ctx, img); err != nil {
+			return err
```

Comment:
Shall we try to store every image?
And aggregate the errors of each store error, and then return the aggregated error?

Reply:
OK. Will update